pghoard/restore.py

```diff
@@ -82,10 +83,19 @@ def __init__(
         size: int,
         new_name: Optional[str] = None,
         metadata: Optional[Dict] = None,
-        file_type: Optional[FileInfoType] = FileInfoType.regular
+        file_type: FileInfoType = FileInfoType.regular
     ):
         self.data: bytes = data
-        super().__init__(size=size, new_name=new_name, metadata=metadata, file_type=file_type)
+        self.metadata = metadata
+        super().__init__(size=size, new_name=new_name, file_type=file_type)
+
+    def __eq__(self, other):
+        return isinstance(other, FileDataInfo) and self.data == other.data and self.size == other.size \
+               and self.new_name == other.new_name and self.metadata == other.metadata \
+               and self.file_type == other.file_type
+
+    def __repr__(self):
+        return f"<data> (size={self.size}, file_type={self.file_type}, metadata={self.metadata})"
```

Comment:
Theses classes with manual `__eq__` and `__repr__` implementations should be dataclasses instead.

Reply:
There is still a limitation in dataclasses inheritance, if you have fields with default values in the base class, then the child class should have all fields with default values. Does not fit very nice.

Reply:
Let me check if I can workaround this.

Reply:
OKish workaround exists from Python 3.10 only (kw_only=True)